### PR TITLE
lightning: fix panic when user cancel

### LINKED ISSUE #41235: lightning may panic when user cancel
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

use failpoint to reproduce https://github.com/pingcap/tiflow/issues/8209

```
panic: send on closed channel

goroutine 345 [running]:
github.com/pingcap/tidb/br/pkg/membuf.(*Pool).release(0xc0006a88d0, {0x7fa9d430b010, 0x100000, 0x100000})
        /home/lance/Projects/tidb/br/pkg/membuf/buffer.go:112 +0x72
github.com/pingcap/tidb/br/pkg/membuf.(*Buffer).Destroy(0xc001718050)
        /home/lance/Projects/tidb/br/pkg/membuf/buffer.go:177 +0x185
github.com/pingcap/tidb/br/pkg/lightning/backend/local.(*Writer).Close(0xc000a494d0, {0x521b7e8?, 0xc000d60d70?})
        /home/lance/Projects/tidb/br/pkg/lightning/backend/local/engine.go:1201 +0x1c3
github.com/pingcap/tidb/br/pkg/lightning/backend.(*LocalEngineWriter).Close(...)
        /home/lance/Projects/tidb/br/pkg/lightning/backend/backend.go:431
github.com/pingcap/tidb/br/pkg/lightning/restore.(*TableRestore).restoreEngine.func3(0x0?, 0xc000dff3a0)
        /home/lance/Projects/tidb/br/pkg/lightning/restore/table_restore.go:595 +0x24e
created by github.com/pingcap/tidb/br/pkg/lightning/restore.(*TableRestore).restoreEngine
        /home/lance/Projects/tidb/br/pkg/lightning/restore/table_restore.go:582 +0x117b
```

### 2. What did you expect to see? (Required)

no panic

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #41235

Problem Summary:

### What is changed and how it works?

exit the function after check goroutine if finished by wg.Done()

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/table_restore.go [hunk 1]
```diff
@@ -524,9 +524,15 @@ func (tr *TableRestore) restoreEngine(
...
+		failpoint.Inject("orphanWriterGoRoutine", func() {
+			if chunkIndex > 0 {
+				<-pCtx.Done()
+			}
+		})
+
 		select {
 		case <-pCtx.Done():
-			return nil, pCtx.Err()
+			break
```
Comment (lichunzhu) [inline br/pkg/lightning/restore/table_restore.go:535]: `break` means break this for loop? This break may not take effect

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
will not break existing lightning integration tests

https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/15109/pipeline/

will merge it

**Pull request merged**